api/v1alpha1/etcdpeer_types.go

```diff
@@ -47,11 +47,11 @@ type EtcdPeerSpec struct {
 	// pods. As per the etcd documentation, etcd will ignore bootstrap
 	// instructions if it already knows where it's peers are.
 	// +optional
-	Bootstrap *Bootstrap `json:"bootstrap,omitempty"`
+	Bootstrap Bootstrap `json:"bootstrap,omitempty"`
```

Comment:
This is a pointer as it is not a required field and is a struct - as per the API conventions doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required):

`They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) or have a built-in nil value (e.g. maps and slices).`

```
Using a pointer allows distinguishing unset from the zero value for that type. There are some cases where, in principle, a pointer is not needed for an optional field since the zero value is forbidden, and thus implies unset. There are examples of this in the codebase. However:

it can be difficult for implementors to anticipate all cases where an empty value might need to be distinguished from a zero value
structs are not omitted from encoder output even where omitempty is specified, which is messy;
having a pointer consistently imply optional is clearer for users of the Go language client, and any other clients that use corresponding types
Therefore, we ask that pointers always be used with optional fields that do not have a built-in nil value.
```

Reply:
It is a required field. See above.